### DISCUSSION TITLE
How to ignore dir or files

### DISCUSSION THREAD

**Discussion by longcp** [BurntSushi/ripgrep#1859] [Q&A] [answered]
![image](https://user-images.githubusercontent.com/10282683/116021085-11377480-a67a-11eb-9a42-cb3b5068d622.png)

use 
rg tcpdump --glob=out or rg tcpdump --glob=!out
would not seach tcpdump without "out" dir,
I have read the README, but still haven’t found a solution, but I know that this requirement must have been fulfilled.

**Comment by BurntSushi** [comment, marked as answer]
Sorry, but I can't understand your question. Please consider saying more precisely what you want to do, the expected output and the actual output. It would be even better if you reproduced your problem on a public repo.

Two other things:

1. Why haven't you used the `--debug` flag?
2. The `!` is used to exclude things. It's that what you want? If so, why aren't you escaping or quoting the `!`?

**Reply by longcp** [reply]
> Sorry, but I can't understand your question. Please consider saying more precisely what you want to do, the expected output and the actual output. It would be even better if you reproduced your problem on a public repo.
> 
> Two other things:
> 
> 1. Why haven't you used the `--debug` flag?
> 2. The `!` is used to exclude things. It's that what you want? If so, why aren't you escaping or quoting the `!`?

Thanks for your reply.
Yes, I want to find out the filed "tcpdump" and hope to exclude the "out" dir.
I tried "rg tcpdump --glob=!out", but it was not work.
now I tried "rg tcpdump --glob='!out' " and "rg tcpdump --glob='!{out,tools}' ", yes, it works, that's what I want, Thanks a lot.
